### PR TITLE
Remove unnecessary wrapper types from config

### PR DESCRIPTION
**Pull request by peterstuart**

### CODE REVIEW

File: src/config/midi.rs [hunk 1]
```diff
@@ -15,51 +15,23 @@ impl Midi {
 
 #[derive(Copy, Clone, Debug, Deserialize)]
 pub struct MidiSlider {
-    pub channel: Channel,
-    pub control_change: ControlChange,
-}
-
-#[derive(Copy, Clone, Debug, Deserialize)]
-#[serde(transparent)]
-pub struct Channel {
```
Comment (peterstuart) [inline src/config/midi.rs:24]: These wrapper structs weren't really necessary. Instead I can just tell serde how to deserialize the `wmidi` types:

```
#[serde(deserialize_with = "deserialize_channel")]
```

File: src/audio/midi.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ pub fn latest_control_value(
 ) -> Option<ControlValue> {
     messages.iter().rev().find_map(|message| match message {
         MidiMessage::ControlChange(ch, function, value)
-            if ch == &*slider.channel && function == &*slider.control_change =>
+            if ch == &slider.channel && function == &slider.control_change =>
```
Comment (helixbass) [inline src/audio/midi.rs:39]: Ok in https://github.com/peterstuart/pedals/pull/23#discussion_r637418782, you removed the `&`'s but here are removing the `*`'s

I'm following the concept of the removal of the wrapper types that were unnecessarily there just for deserialization purposes, but not clear on the relationship of this branch to `tap-tempo` or the resulting merge conflicts

**Review by helixbass — APPROVED** [review]
@peterstuart per comment, this seems like a question of resolving (conceptual/)actual merge conflicts wrt the now-merged `tap-tempo` branch

File: src/audio/midi.rs (hunk 2 above)
Comment (peterstuart) [reply src/audio/midi.rs]: Ok, this is cleaned up now and niether `&` nor `*` are necessary.

**Pull request merged**